VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+                    $vcfCertCa = Get-VCFCertificateAuthority
+                    If ($vcfCertCa.username -ne "$vcfCertCaUsername")
+                    {
+                        Write-Output "Configuring Microsoft Certificate Authority Connection in SDDC Manager using ($vcfCertCaUsername)"
+                        Set-VCFMicrosoftCA -serverUrl $serverUrl -username $vcfCertCaUsername -password $vcfCertCaPassword -templateName $vcfCertCaTemplate | Out-Null
+                        Write-Output "Configuration of Microsoft Certificate Authority in SDDC Manager Using ($($vcfCertCaUsername)): SUCCESSFUL"
+                    } else {
+                        Write-Output "Configuration of Microsoft Certificate Authority in SDDC Manager Using ($($vcfCertCa.username)), already exists: SKIPPED"
+                    }
+                    Write-Output "Finished the Process of Configuring a Microsoft Certificate Authority in SDDC Manager"
+                } 
+                Catch 
+                {
+                    $ErrorMessage = $_.Exception.Message
+                    Write-Output "Error was: $ErrorMessage"
+                }
+            } else {
+                Write-Error "Unable to Authenticate SDDC Manager ($server)"
+            }
+        } else {
+            Write-Error "Unable to connect to Microsoft Certificate Authority ($vcfCertCaFqdn)"
+        }
+    } else {
+        Write-Error "Unable to connect to SDDC Manager ($server)"
+    }
+}
+
+Function gatherSddcInventory {
+    Param (
+        [Parameter (Mandatory = $true)] $domainType,
+        [Parameter (Mandatory = $true)] $workloadDomain
+    )
+
+    # Gathers various deployment details from SDDC Manager
+    $sddcMgr = Get-VCFManager
+    $sddcMgrVersion = $sddcMgr.version.split(".")[0]
+
+    $resourcesObject = @()
+    # SDDC Manager
+    IF ($domainType -eq "Management")
+    {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $sddcMgr.fqdn
+            'name'       = $sddcMgr.fqdn.split(".")[0]
+            'resourceId' = $sddcMgr.id
+            'type'       = "SDDC_MANAGER"
+        } 
+    }
+
+    # vCenters
+    If (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Management"))
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq "MGMT" }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+    elseIf (([float]$sddcMgrVersion -lt [float]4) -AND ($domainType -eq "Workload"))
+    {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq $workloadDomain }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+        $isLegacyWldnsxT = Get-VCFNsxtCluster | Where-Object {$_.domains.id -eq $domain.id}
+        $isLegacyWldnsxV = Get-VCFNsxvManager | Where-Object {$_.domain.id -eq $domain.id}
+    }
+    elseIf (([float]$sddcMgrVersion -ge [float]4) -AND ($domainType -eq "Management"))
+    {
```

Comment:
```suggestion
    } elseIf (([float]$sddcMgrVersion -ge [float]4) -AND ($domainType -eq "Management")) {
```

Reply:
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)